app/src/BoundedContext/BusinessDecision.elm

```diff
@@ -0,0 +1,86 @@
+module BoundedContext.BusinessDecision exposing (..)
+
+import Json.Encode as Encode
+import Json.Decode as Decode
+import BoundedContext exposing (name)
+
+
+type BusinessDecision = 
+    BusinessDecision BusinessDecisionInternal
+    
+type alias BusinessDecisionInternal = 
+    { name : String
+    , description : Maybe String
+    }
+
+type alias BusinessDecisions = List BusinessDecision
+
+type Problem
+  = DefinitionEmpty
+  | AlreadyExists
+
+getName : BusinessDecision -> String 
+getName (BusinessDecision decision) =
+    decision.name
+
+getDescription: BusinessDecision -> Maybe String
+getDescription (BusinessDecision decision) =
+    decision.description
+
+isDecisionNameUnique : String -> BusinessDecisions -> Bool
+isDecisionNameUnique name decisions =
+  decisions
+  |> List.map getName
+  |> List.map String.toLower
+  |> List.member (name |> String.toLower)
+  |> not
+
+defineBusinessDecision : BusinessDecisions -> String -> String -> Result Problem BusinessDecision
+defineBusinessDecision decisions name description =
+    if String.isEmpty name then
+        Err DefinitionEmpty
+    else
+        if isDecisionNameUnique name decisions then
+            Ok <| BusinessDecision <| BusinessDecisionInternal (name |> String.toLower) (if String.isEmpty description then Nothing else Just description)
```

Comment:
Can you reformat this a bit / break it up into multiple lines?
(In this case reversing the order of Ok/BusinessDecision/BusinessDecisionInternal and using `|>` might be helpful)

And I think you should not use `(name |> String.toLower)` for the name, as it would always convert the chosen business name into lowercases (which we probably only want for unique-comparison)

Reply:
makes sense to me. edited